File: sorting/bubbleSort/bubbleSort.java
import java.util.Arrays;

public class bubbleSort {

    static void bubbleSortFunction(int[] arr){

    for(int i=arr.length;i>0;i--){
        Boolean alredySorted = true;
        for(int j=0;j<i-1;j++){
          
            if(arr[j] >arr[j+1]){
                int temp= arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
                alredySorted= false;
            }
            System.out.println(Arrays.toString(arr)   +" One iteration Completed");
        }
        if(alredySorted) break;
        // System.out.println(Arrays.toString(arr)   +" One iteration Completed");
    }

    System.out.println(Arrays.toString(arr)   + " after");
    }

    public static void main(String[] args) {
        int [] arr = {1,4,2,9,23,6,7,90,22,66,77};
        bubbleSortFunction(arr);
    }
    
}

File: namasteDSA/sorting/bubbleSort.js
function bubbleSort(arr){
    let n = arr.length;
    for(let i =0;i<n;i++){
        let swapped = false;
        for(j=0;j<n-1-i;j++){
            if(arr[j]>arr[j+1]){
                let temp = arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp
                swapped = true;
            }
        }
        if(!swapped){
            break
        }
    }
    return arr


}

console.log(bubbleSort([4,7,88,9,1,2,84]))